### DISCUSSION TITLE
UI Components suggestions

### DISCUSSION THREAD

**Discussion by flowsn4ke** [tailwindlabs/tailwindcss#2903] [Ideas]
👋 Hello there!

Having adopted Tailwind a week ago I obviously jumped on the occasion to purchase both UI packages, which I couldn't be more happy with! **It saves us a ton of time both in prototyping and developing stages** ⚡

I know there are plans to add much more and **I would like to suggest a few components ideas** that would really come in handy from my perspective.

Maybe you already have plans for those, maybe not, here it is anyway:

- **Range Sliders input** - this one is self-explanatory
- **Upload Progress Bars** - think file uploading stuff
- **Sidebars with dropdown sections** - for menu items
- **Accordeon cards** - I'm thinking something around the lines of the design you can see [at the very bottom of this homepage](https://www.renkema.com/)
- **Photo galleries** - huge pain point doing it over and over again, would be neat to have a clean starting point
- **Product galleries** - e-commerce stuff, some 'basic' cards with image, name, price tag, add to cart button (maybe also quantity?)
- **Search fields** - with icons and maybe a dropdown to display a range of clickable search results
- **Aspect Ratio container** that keeps the shape of it's child elements, I remember the solution for that involves doing some weird math and setting negative margins, I have to google for it every time and it takes me forever, and I'm pretty sure I'm not the only one. That would allow to keep a video or image centered in a page no matter its aspect ratio
- **Charts**, if doable at all in pure html/css with some JS input obviously, I have little experience with those but it would come in VERY handy
- **Animated loaders**, for page / component loading

🙏 Thanks for considering giving it a shot! Keep the amazing stuff coming!

**Comment by aricma** [comment]
@theophilelouvel great list.
I would like to add **`tooltips`** as a component idea

**Comment by DebraHird** [comment]
Input Controls: checkboxes, radio buttons, dropdown lists, list boxes, buttons, toggles, text fields, date field.
Navigational Components: breadcrumb, slider, search field, pagination, slider, tags, icons.
Informational Components: tooltips, icons, progress bar, notifications, message boxes, modal windows.

[My Great Lakes](https://www.mygreatlakes.me/)
